### ISSUE TITLE
Questions on  implementation of custom_loss

### ISSUE THREAD

**Issue by shikiponn** [experiencor/basic-yolo-keras#198]
I went over your codes of custom_loss, but still don't understand several parts of it. I'm new to TF, so some questions might be obvious to you but please help me. Thank you in advance.

1. Why `seen = tf.Variable(0.)` is defined in the loss function? It seems that it always set to 0 everytime when you call the loss function, but should work as a counter.  Why it is not defined outside the loss function?

2. Related to the question above, I don't understand what's going on the line 198~200 in frontend.py.  In Readme,  you wrote
``` # the number of initial epochs during which the sizes of the 5 boxes in each cell is forced to match the sizes of the 5 anchors, this trick seems to improve precision emperically```
However, I don't see how your implementation does the trick like that.

3.  I don't see what the mask below is going to mask, which is from line 194 in frontend.py.
`no_boxes_mask = tf.to_float(coord_mask < COORD_SCALE/2.)`

**Comment by experiencor** [comment]
Hi @shikiponn,

1. Please note line 195 of frontend.py. It's possible to define outside the loss function.

2. Please read line 197 - 203 carefully.

3. It masks x, y, w, and h.

**Comment by shikiponn** [comment]
Thank you @experiencor!
Now I'm clear for my question 1 and 3.
For question 2, I will check your code again!
